source/puppet/5.1/lang_data_hash.md

```diff
@@ -29,17 +29,23 @@ notice($result)
 
 ## Syntax
 
-Hashes are written as a pair of curly braces containing any number of key/value pairs. A key is separated from its value by a `=>` (arrow, fat comma, or hash rocket), and adjacent pairs are separated by commas. An optional trailing comma is allowed between the final value and the closing curly brace.
+Hashes are written as a pair of curly braces containing any number of key/value pairs. Keys should be string encapsulated. A key is separated from its value by a `=>` (arrow, fat comma, or hash rocket), and adjacent pairs are separated by commas. An optional trailing comma is allowed between the final value and the closing curly brace.
```

Comment:
I have issues with the term "string encapsulated" instead of something like "should be quoted with single or double quote". In addition it is somewhat confusing since any type of key is allowed (if you want to you can use say an Array, and then it should not be "quoted" or turned into string form unless it is a requirement where the hash will be used that it must be a String. (Will expand on this in a general comment).